alicloud/resource_alicloud_alikafka_instance.go

```diff
@@ -0,0 +1,332 @@
+	zoneId := d.Get("zone_id").(string)
+
+	// 1. Create post-pay order
+	createOrderReq := alikafka.CreateCreatePostPayOrderRequest()
+	createOrderReq.RegionId = regionId
+	createOrderReq.TopicQuota = requests.NewInteger(topicQuota)
+	createOrderReq.DiskType = strconv.Itoa(diskType)
+	createOrderReq.DiskSize = requests.NewInteger(diskSize)
+	createOrderReq.DeployType = requests.NewInteger(deployType)
+	createOrderReq.IoMax = requests.NewInteger(ioMax)
+	if v, ok := d.GetOk("eip_max"); ok {
+		createOrderReq.EipMax = requests.NewInteger(v.(int))
+	}
+
+	var createOrderResp *alikafka.CreatePostPayOrderResponse
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+			return alikafkaClient.CreatePostPayOrder(createOrderReq)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+				time.Sleep(10 * time.Second)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(createOrderReq.GetActionName(), raw, createOrderReq.RpcRequest, createOrderReq)
+		v, _ := raw.(*alikafka.CreatePostPayOrderResponse)
+		createOrderResp = v
+		return nil
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_alikafka_instance", createOrderReq.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	alikafkaInstanceVO, err := alikafkaService.DescribeAlikafkaInstanceByOrderId(createOrderResp.OrderId, 60)
+
+	if err != nil {
+		return WrapError(err)
+	}
+
+	instanceId := alikafkaInstanceVO.InstanceId
+	d.SetId(instanceId)
+
+	// 3. Start instance
+	startInstanceReq := alikafka.CreateStartInstanceRequest()
+	startInstanceReq.RegionId = regionId
+	startInstanceReq.InstanceId = instanceId
+	startInstanceReq.VpcId = vpcId
+	startInstanceReq.VSwitchId = vswtichId
+	startInstanceReq.ZoneId = zoneId
+	if _, ok := d.GetOk("eip_max"); ok {
+		startInstanceReq.IsEipInner = requests.NewBoolean(true)
+		startInstanceReq.DeployModule = "eip"
+	}
+	if v, ok := d.GetOk("name"); ok {
+		startInstanceReq.Name = v.(string)
+	}
+
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		raw, err := alikafkaService.client.WithAlikafkaClient(func(alikafkaClient *alikafka.Client) (interface{}, error) {
+			return alikafkaClient.StartInstance(startInstanceReq)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, []string{AlikafkaThrottlingUser}) {
+				time.Sleep(10 * time.Second)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(startInstanceReq.GetActionName(), raw, startInstanceReq.RpcRequest, startInstanceReq)
+		return nil
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_alikafka_instance", startInstanceReq.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	// 3. wait until running
+	err = alikafkaService.WaitForAlikafkaInstance(d.Id(), Running, DefaultLongTimeout)
```

Comment:
please use timeouts to do it. Like https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_instance.go#L394